addons/mail/static/src/discuss/core/common/message_service_patch.js

```diff
@@ -0,0 +1,20 @@
+/* @odoo-module */
+
+import { MessageService } from "@mail/core/common/message_service";
+
+import { patch } from "@web/core/utils/patch";
+patch(MessageService.prototype, {
+    async edit(
+        message,
+        body,
+        attachments = [],
+        { mentionedChannels = [], mentionedPartners = [] } = {}
+    ) {
+        const fetchedChannels = await Promise.all(message.mentionedChannelPromises);
+        const allChannels = this.store.Thread.insert([...fetchedChannels, ...mentionedChannels]);
```

Comment:
Undefineds should be filtered here, regarding channels that no longer exists (or no longer accessible).

We should also have a test to add coverage of invalid model or ids.